Wrong use of return type "NoReturn"

thunfisch987 opened the pull request:
### Description of the problem

In a lot of the type hinted functions, the return type hint "NoReturn" was wrongly used.
It's supposed to be used with functions that (under all conditions) _do not_ return.
Example for a NoReturn function:
```python
def stop() -> NoReturn:
    raise RuntimeError()
```

### Reproducing the problem

An Example of where it was used wrongly:
```python
def set_active_underline_width(self, width: Union[float, int]) -> NoReturn:
    """Animates the width of the active underline line."""

    Animation( _underline_width=width, duration=(0.2 if self.line_anim else 0), t="out_quad" ).start(self)
```
[Animation(...).start() always returns None] 

### Description of Changes

Every occasion of NoReturn was checked and changed to None, as all of them were wrong.

thunfisch987 committed:
Wrong use of return type "NoReturn"
in all of the occasions, None should be used as the return
there was also a type hint which said NoReturn instead of float

thunfisch987 commented:
there's also a wrong type hint in datatables.py which said `NoReturn` which i changed to `None`, but should be `bool`

HeaTTheatR commented on kivymd/toast/kivytoast/kivytoast.py:
```diff
@@ -38,7 +38,7 @@ def build(self):
 """
 
-from typing import List, NoReturn
+from typing import List, None
```
```python
from typing import List, None
```

```
  File "<input>", line 1
    from typing import List, None
                             ^
SyntaxError: trailing comma not allowed without surrounding parentheses
```

HeaTTheatR commented on kivymd/tools/patterns/create_project.py:
```diff
@@ -228,37 +228,37 @@ def __init__(self, model):
 '''
 
-_firebase_view_import = """from typing import Union, NoReturn
+_firebase_view_import = """from typing import Union, None
```
```python
from typing import List, None
```

```
  File "<input>", line 1
    from typing import List, None
                             ^
SyntaxError: trailing comma not allowed without surrounding parentheses
```

HeaTTheatR commented on kivymd/uix/behaviors/backgroundcolor_behavior.py:
```diff
@@ -7,7 +7,7 @@
 __all__ = ("BackgroundColorBehavior", "SpecificBackgroundColorBehavior")
 
-from typing import List, NoReturn
+from typing import List, None
```
```python
from typing import List, None
```

```
  File "<input>", line 1
    from typing import List, None
                             ^
SyntaxError: trailing comma not allowed without surrounding parentheses
```

HeaTTheatR commented on kivymd/uix/filemanager/filemanager.py:
```diff
@@ -133,7 +133,7 @@ def file_manager_open(self):
 import os
 import re
-from typing import List, NoReturn, Tuple, Union
+from typing import List, None, Tuple, Union
```
```python
from typing import List, None
```

```
  File "<input>", line 1
    from typing import List, None
                             ^
SyntaxError: trailing comma not allowed without surrounding parentheses
```

HeaTTheatR commented on kivymd/uix/pickers/colorpicker/colorpicker.py:
```diff
@@ -14,7 +14,7 @@
 .. code-block:: python
 
-    from typing import Union, NoReturn
+    from typing import Union, None
```
```python
from typing import List, None
```

```
  File "<input>", line 1
    from typing import List, None
                             ^
SyntaxError: trailing comma not allowed without surrounding parentheses
```

HeaTTheatR commented on kivymd/uix/pickers/colorpicker/colorpicker.py:
```diff
@@ -76,7 +76,7 @@ def on_select_color(self, instance_gradient_tab, color: list) -> NoReturn:
 import struct
 from io import BytesIO
-from typing import List, NoReturn, Union
+from typing import List, None, Union
```
```python
from typing import List, None
```

```
  File "<input>", line 1
    from typing import List, None
                             ^
SyntaxError: trailing comma not allowed without surrounding parentheses
```

thunfisch987 commented:
Oh....
Looks like there were some imports i overlooked😬
(Since this is my first commit, how do i fix it? Do i open a new one with the fixed code?)

HeaTTheatR commented:
@thunfisch987 Make another push.

thunfisch987 committed:
forgot to run pre-commit, flake8 errors fixed

thunfisch987 committed:
removed None from typing imports
- bool return (mentioned in
https://github.com/kivymd/KivyMD/pull/1186#issuecomment-1049925477)
fixed

HeaTTheatR commented on kivymd/uix/templates/scalewidget/scalewidget.py:
```diff
@@ -15,7 +15,7 @@
 .. code-block:: python
 
-    
+
```
There must be only one newline

thunfisch987 replied:
sorry, forgot that pre-commit does not check the docstrings

HeaTTheatR commented on kivymd/uix/templates/rotatewidget/rotatewidget.py:
```diff
@@ -15,7 +15,7 @@
 
-    
+
 
```
There must be only one newline

thunfisch987 committed:
fix only one newline

thunfisch987 commented on kivymd/tools/patterns/create_project.py:
```diff
@@ -179,7 +179,7 @@ class {name_screen}Model(BaseScreenModel):
 
-_firebase_controller = '''from typing import NoReturn
+_firebase_controller = '''
 {import_module}
```
is this still needed?

thunfisch987 replied:
i mean the {import_module}

thunfisch987 replied:
because i don't know anything about firebase

thunfisch987 commented on kivymd/tools/patterns/create_project.py:
```diff
@@ -197,22 +197,22 @@ def __init__(self, model):
 
-_without_firebase_controller = '''from typing import NoReturn
+_without_firebase_controller = '''
 {import_module}
```
and this one

thunfisch987 committed:
newline fix number 2
